sdk/iot/modelsrepository/src/fetcherFilesystem.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft.
+// Licensed under the MIT license.
+
+import fs from "fs";
+import * as path from "path";
+import { Fetcher, DTDL, logger } from "./internal";
+import {RestError, RestErrorOptions} from '@azure/core-rest-pipeline';
+
+export class FilesystemFetcher extends Fetcher {
+  private _baseFilePath: string;
+
+  constructor(baseFilePath: string) {
+    super();
+    this._baseFilePath = baseFilePath;
+  }
+
+  fetch(filePath: string) {
+    logger.info(`Fetching ${filePath} from local filesystem`);
+    const absolutePath = path.join(this._baseFilePath, filePath);
+
+    try {
+      logger.info(`File open on ${absolutePath}`);
+      const dtdlFile = fs.readFileSync(absolutePath, "utf8");
```

Comment:
Using `sync` methods is generally considered bad practice since it blocks the main thread. You could use the promise version of readFile from `fs/promise`, which would end up looking very similar if you also make `fetch` async and await readFile.

Edit: Looks like fs/promise was added in node 10, so if you support older versions of node than that, you'd need to use fs.readFile which accepts a callback. In that case it might be worth writing a simple wrapper around readFile that returns a promise.